contracts/Everlong.sol

```diff
@@ -215,50 +239,95 @@ contract Everlong is IEverlong {
             );
     }
 
-    /// @dev Rebalance after a deposit if needed.
+    /// @dev Attempt rebalancing after a deposit if idle is above max.
     function _afterDeposit(uint256, uint256) internal virtual override {
-        if (canRebalance()) {
+        if (ERC20(_asset).balanceOf(address(this)) > maxIdleLiquidity()) {
             rebalance();
         }
     }
 
     /// @dev Frees sufficient assets for a withdrawal by closing positions.
-    /// @param assets Amount of assets owed to the withdrawer.
+    /// @param _assets Amount of assets owed to the withdrawer.
     function _beforeWithdraw(
-        uint256 assets,
+        uint256 _assets,
         uint256
     ) internal virtual override {
-        // Close more positions until sufficient idle to process withdrawal.
-        _closePositions(assets - ERC20(_asset).balanceOf(address(this)));
+        // Close more positions until sufficient idle to process withdrawal
+        // and keep remaining target.
+        _closePositions(
+            _assets -
+                ERC20(_asset).balanceOf(address(this)) +
+                targetIdleLiquidity()
```

Comment:
Won't we underflow if `_assets < ERC20(_asset).balanceOf(address(this))`?

Reply:
Also, what happens if `ERC20(_asset).balanceOf(address(this))` is greater than the target. Let's say that the target percentage is 10%, the max percentage is 20%, and the `totalAssets` is 10 million. Let's also say that `_assets` is 10 thousand, and `ERC20(_asset).balanceOf(address(this))` is 1.5 million. 

This calculation (ignoring underflows) will give us:

```
10_000 - 1_500_000 + 1_000_000 = -490_000
```

That value really doesn't make sense in this context. In this case, we should be clamping to zero.

Reply:
On second thought not even sure if it makes sense to close positions to bring up idle during a withdrawal since the whole point of idle imo is to service withdrawals without closing positions.... thinking of just removing idle consideration from withdrawal path

Reply:
agreed externally that the above plan is okay